api/Controllers/AccountsController.cs

```diff
@@ -56,7 +63,53 @@ public async Task<IActionResult> GetAccountEmail([FromRoute] int Id)
             return Ok(email);
         }
 
+        [Route("{id}")]
+        [HttpPut]
+        [Authorize]
+        public async Task<IActionResult> ChangeAccountPassword([FromBody] ChangePassword request)
+        {
+            if (request.NewPassword == null)
+            {
+                return BadRequest("No new password");
+            }
+
+            string authorizationHeaderValue = Request.Headers["Authorization"].ToString();
+            int id = _jwtService.GetAccountId(authorizationHeaderValue);
+
+            AccountEntity account = await GetAccountById(id);
+
+            if (account == null)
+            {
+                return BadRequest("Such account does not exist");
+            }
```

Comment:
Is it even possible? You have `[Authorize]` for that method. How does `[Authorize]` work?

Reply:
If this method is available, I am precisely authorized. So I should not check it one more time
About you question, I get token from request and see if it satisfies the requirements (from startup file). If yes so user is authorized and method is available

Reply:
Have discussed it; if authorized then account exists for sure